### DISCUSSION TITLE
get LAB values

### DISCUSSION THREAD

**Discussion by SylvainArd** [ImageMagick/ImageMagick#5054] [Help]
Hello I have an ICC profile v2 for my scanner and a RGB TIFF image scanned by my scanner. What I want is to create a LAB TIFF image from the original RGB TIFF image according to my ICC profile v2. What is the command please ?
Thank you !
Best regards

**Comment by snibgo** [comment]
If I understand correctly, you have a RGB TIFF from your scanner with embedded profile. You want to create a file encoded in CIELab.
```
magick in.tiff -profile some_lab_profile.icc -define quantum:format=floating-point -depth 32 out.tiff
```
For "some_lab_profile.icc", I suggest "Lab-D50-Identity-elle-V2.icc" from https://github.com/ellelstone/elles_icc_profiles/tree/master/profiles . Elle has a description of her profiles at https://ninedegreesbelow.com/photography/lcms-make-icc-profiles.html .

**Comment by SylvainArd** [comment]
no the TIFF has not an embedded profile I have the ICC profile v2 apart
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
sorry I don't know if all have received it so I post here :
no the TIFF has not an embedded profile I have the ICC profile v2 apart
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
and even if the original RGB TIFF image has an embeded ICC profile I want
to force it to use mine apart
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
so the good answer is :

magick in.tiff -profile some_lab_profile.icc -define
quantum:format=floating-point -depth 32 out.tiff

?
and it recognizes ICC profiles v2 ?
and how to get the CieLab value of a certain pixel in the resulting TIFF
please ?
Thank you very much

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
Okay, the profile is known but is not embedded. So you need to assign that profile first:
```
magick in.tiff -profile scanner_profile.icc -profile some_lab_profile.icc -define quantum:format=floating-point -depth 32 out.tiff
```

If in.tiff has a profile, but the wrong one, remove all ICC profiles first:
```
magick in.tiff +profile ICC -profile scanner_profile.icc -profile some_lab_profile.icc -define quantum:format=floating-point -depth 32 out.tiff
```

"-define quantum:format=floating-point -depth 32" will save the output as 32 bit/channel floating point. That may be overkill for you.

**Comment by SylvainArd** [comment]
what is the difference between scanner_profile.icc and some_lab_profile.icc,
I have only one ICC profile ?!
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> and it recognizes ICC profiles v2 ?

ImageMagick can use v2 or v4 profiles.

> and how to get the CieLab value of a certain pixel in the resulting TIFF please ?

I suggest:
```
magick out.tiff -crop 1x1+12+34 +repage ftxt:

0,0:2.1138304e+09,4.2252069e+09,3.3691955e+08
```
This "-crop" makes a one-pixel image, from the pixel at (12,34). "ftxt:" gives the value in each channel, on a nominal scale from 0 to QuantumRange. I use IM Q32, QuantumRange is 2^32-1 which is 4294967295. If you use Q16, then your QuantumRange is 65535.

> what is the difference between scanner_profile.icc and some_lab_profile.icc, I have only one ICC profile ?!

What is the name of the ICC profile you have? That is the profile for the scanner, is it? Then use that name instead of scanner_profile.icc

Instead of some_lab_profile.icc, use any Lab profile you want. I suggest the one I linked to, Lab-D50-Identity-elle-V2.icc, or the V4 version if you prefer.

**Comment by SylvainArd** [comment]
it's my last question + is imagemagick is supporting ICC profiles v2
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> it's my last question + is imagemagick is supporting ICC profiles v2

Yes. ImageMagick can use v2 or v4 profiles.

**Comment by SylvainArd** [comment]
thank you very much, one last question : what's the goal of the
some_lab_profile.icc
and why there are several ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
if I understand the command :

magick in.tiff +profile ICC -profile scanner_profile.icc -profile
some_lab_profile.icc -define quantum:format=floating-point -depth 32
out.tiff

remove all ICC profiles in in.tiff and add scanner_profile.icc and
some_lab_profile.icc
in in.tiff in the same operation ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> remove all ICC profiles in in.tiff and add scanner_profile.icc and some_lab_profile.icc in in.tiff in the same operation ?

It is one command.

The command does not "add" profiles. Profiles are not "added".

+profile ICC  <== this removes any embedded ICC profiles.

-profile scanner_profile.icc <== this assigns a profile, by embedding the profile as metadata.

-profile some_lab_profile.icc <== this changes pixels values to convert them from the embedded profile to the new profile, removes the old profile from metadata, and embeds the new profile as metadata.

**Comment by SylvainArd** [comment]
so if I understand I have to do only the following command :
magick in.tiff +profile ICC -profile scanner_profile.icc -profile
some_lab_profile.icc -define quantum:format=floating-point -depth 32
out.tiff
and I have all I want isn't it ?

Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
thank you very much
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
and what is quantum, with crop I don't get the exact LAB values isn't it ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
another question how to get a rectangle of LAB values instead of a single
pixel please ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
> how to get a rectangle of LAB values instead of a single pixel please ?

Above, I showed "`-crop 1x1+12+34`" to get a single pixel. The four numbers are WxH+X+Y where W is the width of the rectangle you want, H is the height, X and Y are the offsets of the top-left of the rectangle.

**Comment by SylvainArd** [comment]
Ok I found the solution before
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers
